File: results/local/2026-09-09/current-fleet-composition-v628/viewer/verify-http.mjs
import assert from 'node:assert/strict';
import { readFile, writeFile } from 'node:fs/promises';
import { createHash } from 'node:crypto';
import { dirname, resolve, join } from 'node:path';
import { fileURLToPath } from 'node:url';
import { spawnSync } from 'node:child_process';

const kit = dirname(fileURLToPath(import.meta.url));
const viewer = resolve(kit, '../../../results/lambda/2026-09-06/visualiser');
const digest = b => createHash('sha256').update(b).digest('hex');
const stages = [];
for (const [name, args] of [['schema', ['scripts/check.mjs']], ['syntax', ['--check', 'app.js']]]) {
  const result = spawnSync(process.execPath, args, { cwd: viewer, encoding: 'utf8', timeout: 60000 });
  await writeFile(join(kit, `${name}.log`), result.stdout + result.stderr, { flag: 'wx' });
  stages.push({ name, command: ['node', ...args], exit_code: result.status });
  assert.equal(result.status, 0);
}
const paths = ['index.html', 'app.js', ...['manifest.json', 'fleet.json', 'compute.json', 'checker-binding.json'].map(p => `data/gtoc12-current-composition-v628/${p}`)];
const checks = [];
for (const path of paths) {
  const response = await fetch(`http://127.0.0.1:4173/${path}`, { signal: AbortSignal.timeout(10000) });
  assert.equal(response.status, 200);
  const bytes = Buffer.from(await response.arrayBuffer());
  assert.equal(digest(bytes), digest(await readFile(join(viewer, path))));
  checks.push({ path, status: response.status, bytes: bytes.length, sha256: digest(bytes) });
}
const app = await readFile(join(viewer, 'app.js'), 'utf8');
const html = await readFile(join(viewer, 'index.html'), 'utf8');
assert(app.includes('const initialDataset = availableFleets.has(datasetParam) ? datasetParam : "gtoc12-current-composition-v628"'));
assert(html.includes('<option value="gtoc12-current-composition-v628" selected disabled>Local fleet composition v628'));
assert(!html.includes('<option value="gtoc12" selected'));
for (const retained of ['gtoc12-regeneration-v799', 'gtoc12-seeded-candidate-v627']) {
  assert(app.includes(retained) && html.includes(retained));
}
assert(!app.split('\n').find(line => line.includes('"gtoc12-orphan-v595":')).includes('best'));
const report = {
  status: 'passed', checked_utc: new Date().toISOString(), http_checks: checks, stages,
  url: 'http://127.0.0.1:4173/?dataset=gtoc12-current-composition-v628&ship=8&epoch=69807&preset=oblique&z=1',
  default_dataset: 'gtoc12-current-composition-v628',
  H100_v799_and_historical_v627_retained: true, stale_v595_best_label_removed: true,
  independent_history_audit_sha256: 'd22bd60ba92d5caf816fefd9b82acaafb2d0fec4b2b05c6270f206d6fa99f37c',
  server: 'Existing localhost:4173 server returned exact local bytes; no server was started or stopped.',
  browser_visual_check: 'HTTP and data/schema verification only; parent will open the user-facing URL.',
};
await writeFile(join(kit, 'http-audit.json'), JSON.stringify(report, null, 2) + '\n', { flag: 'wx' });
console.log(JSON.stringify({ status: report.status, http_paths_verified: checks.length, default_dataset: report.default_dataset }));

File: results/local/2026-09-09/frontier-viewer-v629/verify_http.py
"""Check exact locally served display bytes; no solver or propagation."""
import hashlib
import json
from datetime import datetime, timezone
from pathlib import Path
from urllib.request import urlopen

ROOT = Path(__file__).resolve().parents[2]
VIEWER = ROOT / "results/lambda/2026-09-06/visualiser"
OUTPUT = ROOT / "results/local/2026-09-09/frontier-viewer-v629"
DATASET = "gtoc12-frontier-v629"


def digest(raw):
    return hashlib.sha256(raw).hexdigest()


def main():
    if not __debug__:
        raise RuntimeError("Assertions must be enabled")
    OUTPUT.mkdir(exist_ok=False)
    paths = ["index.html", "app.js", "gtoc12.js", "replay.js", "kepler.js"]
    paths += [f"data/{DATASET}/{p}" for p in (
        "manifest.json", "fleet.json", "compute.json", "checker-binding.json", "saved-node-provenance.json")]
    checks = []
    for name in paths:
        local = (VIEWER / name).read_bytes()
        with urlopen(f"http://127.0.0.1:4173/{name}", timeout=10) as response:
            actual = response.read()
            assert response.status == 200 and actual == local
            checks.append({"path": name, "status": response.status, "bytes": len(actual), "sha256": digest(actual)})
    app = (VIEWER / "app.js").read_text(encoding="utf-8")
    html = (VIEWER / "index.html").read_text(encoding="utf-8")
    assert f'const initialDataset = availableFleets.has(datasetParam) ? datasetParam : "{DATASET}"' in app
    assert f'<option value="{DATASET}" selected disabled>Latest verified fleet' in html
    for name in ("gtoc12-current-union-v629", "gtoc12-collect-composition-v807", "gtoc12-current-composition-v628"):
        assert name in app
    manifest = json.loads((VIEWER / f"data/{DATASET}/manifest.json").read_bytes())
    assert manifest["source"]["solution_sha256"] == "765cb7ef97d38926317dbbf4ae8700626dffec06c48af3d6f68dae47c154f3da"
    assert manifest["summary"]["ships"] == 23 and manifest["summary"]["replay_points"] == 11652
    report = {
        "passed": True, "utc": datetime.now(timezone.utc).isoformat(), "http_assets": checks,
        "default_dataset": DATASET, "solution_sha256": manifest["source"]["solution_sha256"],
        "scope": "Exact HTTP bytes and registration; no new propagation or GPU work",
        "prior_verification": {
            "node_tests": {"passed": 16, "failed": 0, "skipped": 1,
                           "skip_reason": "Historical fleet_master_v1 source export unavailable",
                           "files": ["tests/replay.test.mjs", "tests/geometry.test.mjs", "tests/gtoc12-import.test.mjs"]},
            "schema_check": "node scripts/check.mjs: exit 0; all installed datasets validate",
            "independent_read_only_review": "Gap rendering and 142 source/data hashes approved; unchanged 21 histories and original bindings verified",
            "browser": {"tab_id": "31", "url": f"http://127.0.0.1:4173/?dataset={DATASET}&ship=8&epoch=69807&preset=oblique&z=1",
                        "observed_accessibility_state": "Correct selected dataset, 23 ships/199 asteroids, source label, ship8 missing intervals2, Active WebGL2 on RTX5090",
                        "pixel_screenshot_review": False},
        },
    }
    source_paths = ["app.js", "gtoc12.js", "index.html", "replay.js", "scripts/import-gtoc12.mjs", "tests/replay.test.mjs"]
    report["reviewed_source"] = {name: digest((VIEWER / name).read_bytes()) for name in source_paths}
    (OUTPUT / "verify_http.py").write_bytes(Path(__file__).read_bytes())
    (OUTPUT / "http-and-validation.json").write_text(json.dumps(report, indent=2) + "\n", encoding="utf-8")
    (OUTPUT / ".gitattributes").write_text("* -text -whitespace\n", encoding="utf-8")
    print(json.dumps({"passed": True, "http_assets": len(checks), "output": str(OUTPUT)}))


if __name__ == "__main__":
    main()
